UI Revamp 2/n: Implement basic views

amyrebecca opened the pull request:
Get basic views working for CRUDding subject rules, user rules, subject rule effects, and user rule effects

amyrebecca commented:
Part of #363

marten commented on app/views/subject_rules/_form.html.erb:
```diff
@@ -0,0 +1,11 @@
+  <%= f.error_notification %>
+
+  <div class="form-inputs">
+    <%= f.input :condition_string, label: 'Condition' %>
```
Wouldn't this send it in under the `condition_string` key in the params, and then thus be filtered out?

amyrebecca replied:
yeah it probably would, i didn't remember that you had updated that in the controller.

zwolf commented on app/models/effects/add_subject_to_collection.rb:
```diff
@@ -14,6 +14,10 @@ def valid?
     end
 
+    def self.config_fields
+      ["collection_id"]
```
Config fields (here and in other models) should be frozen and probably constants

zwolf commented on app/controllers/user_rules_controller.rb:
```diff
@@ -11,50 +11,57 @@ def index
     authorize workflow
     rule = workflow.user_rules.find(params[:id])
 
+    UserRule.transaction do
```
Doable automatically with `dependent: destroy`

zwolf commented on app/controllers/user_rules_controller.rb:
```diff
@@ -11,50 +11,57 @@ def index
+      rule.destroy
+    end
+
     rule.destroy
```
double destruction

amyrebecca commented:
@zwolf @marten I think I've addressed the concerns you raised--thank you for the review!--so let me know if this is ready to merge! i merged the updates to the other PR in this chain as well
